Sources/Build/BuildPlan/BuildPlan.swift

```diff
@@ -654,6 +662,34 @@ public class BuildPlan: SPMBuildCore.BuildPlan {
         }
         return try description.symbolGraphExtractArguments()
     }
+
+    /// Returns the files and directories that affect the build process of this build plan.
+    public func inputs() -> [Input] {
+        var inputs: [Input] = []
+        for package in self.graph.rootPackages {
+            inputs += package.modules
```

Comment:
Does this include outputs of command plugins?

Reply:
Isn’t this what the following line a little further down is for? Or am I missing something?

```swift
inputs.append(contentsOf: derivedSourceDirPaths.sorted().map { .directoryStructure($0) })
```

Reply:
I've put another comment on relevant lines where context for it is more obvious.